src/benchmark/file_io_write_micro_benchmark.cpp

```diff
@@ -1,84 +1,37 @@
-    }
-    chmod(filename, S_IRUSR | S_IWUSR);  // enables owner to read and write file
-  }
-
-  void TearDown(::benchmark::State& /*state*/) override {
-    Assert(std::remove(filename) == 0, "Remove error: " + std::strerror(errno));
+void write_data_using_write(const size_t from, const size_t to, int32_t fd, uint32_t* data_to_write_start) {
+  const auto uint32_t_size = ssize_t{sizeof(uint32_t)};
+  const auto bytes_to_write = static_cast<ssize_t>(uint32_t_size * (to - from));
+  lseek(fd, from * uint32_t_size, SEEK_SET);
+  if (write(fd, data_to_write_start + from, bytes_to_write) != bytes_to_write) {
+    close(fd);
+    Fail("Write error:" + std::strerror(errno));
   }
+}
 
- protected:
-  std::vector<uint32_t> data_to_write;
-  uint64_t control_sum = uint64_t{0};
-  uint32_t vector_element_count;
-  uint32_t VALUE_TO_WRITE = 42;
-  const char* filename = "file.txt";  //const char* needed for C-System Calls
-
-  void mmap_write_benchmark(benchmark::State& state, const int flag, const int data_access_mode, const ssize_t file_size);
-  void sanity_check(uint32_t NUMBER_OF_BYTES);
-};
-
-void FileIOWriteMicroBenchmarkFixture::sanity_check(uint32_t NUMBER_OF_BYTES) {
-  auto fd = int32_t{};
-  if ((fd = open(filename, O_RDONLY)) < 0) {
+void write_data_using_pwrite(const size_t from, const size_t to, int32_t fd, uint32_t* data_to_write_start) {
+  const auto uint32_t_size = ssize_t{sizeof(uint32_t)};
+  const auto bytes_to_write = static_cast<ssize_t>(uint32_t_size * (to - from));
+  if (pwrite(fd, data_to_write_start + from, bytes_to_write, from * uint32_t_size) != bytes_to_write) {
     close(fd);
-    Fail("Open error:" + std::strerror(errno));
+    Fail("Write error:" + std::strerror(errno));
   }
-
-
-  const auto file_size = lseek(fd, 0, SEEK_END);
-  Assert(file_size == NUMBER_OF_BYTES, "Sanity check failed: Actual size of " + std::to_string(file_size) +
-      " does not match expected file size of " + std::to_string(NUMBER_OF_BYTES) + ".");
-
-  auto read_data = std::vector<uint32_t>(NUMBER_OF_BYTES / sizeof(uint32_t));
-
-  const off_t OFFSET = 0;
-  auto* map = reinterpret_cast<uint32_t*>(mmap(NULL, NUMBER_OF_BYTES, PROT_READ, MAP_PRIVATE, fd, OFFSET));
-  close(fd);
-
-  Assert(map != MAP_FAILED, "Mapping for Sanity Check Failed:" + std::strerror(errno));
-
-  memcpy(std::data(read_data), map, NUMBER_OF_BYTES);
-  const auto sum = std::accumulate(read_data.begin(), read_data.end(), uint64_t{0});
-  Assert(control_sum == sum,
-         "Sanity check failed. Got: " + std::to_string(sum) + " Expected: " + std::to_string(control_sum));
-
-  // Remove memory mapping after job is done.
-  Assert(munmap(map, NUMBER_OF_BYTES) == 0, "Unmapping for Sanity Check failed: " + std::strerror(errno));
 }
 
-BENCHMARK_DEFINE_F(FileIOWriteMicroBenchmarkFixture, WRITE_NON_ATOMIC)(benchmark::State& state) {
+void FileIOWriteMicroBenchmarkFixture::write_non_atomic_single_threaded(benchmark::State& state) {
```

Comment:
We can use the new method 'write_data_using_write' here for consistency and to reduce code duplication.

Reply:
yes exactly. I mean it's a suggestion, i'm fine with both alternatives.

Reply:
skipped for now, future todo - this way its consistent with write